From 71de409709be843b4072f657503afa652263f37c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Joachim=20Mairb=C3=B6ck?= <j.mairboeck@gmail.com>
Date: Tue, 18 Aug 2026 13:42:10 +0200
Subject: Build drivers with clones with the install rpath

Otherwise the clones retain their build rpath because they are not installed
as targets but as files.

diff --git a/cmake/robot.cmake b/cmake/robot.cmake
index 1e6b270..4e6d097 100644
--- a/cmake/robot.cmake
+++ b/cmake/robot.cmake
@@ -181,6 +181,9 @@ MACRO(ROBOT_MODULE)
     ENDIF()
     SET(MODLOC "${MODLOC}/${RBM_NAME}${CMAKE_SHARED_MODULE_SUFFIX}")
 
+    # build with the install rpath because otherwise it is not correctly changed for clones
+    SET_TARGET_PROPERTIES(${RBM_NAME} PROPERTIES BUILD_WITH_INSTALL_RPATH TRUE)
+
     FOREACH(CLONENAME ${RBM_CLONENAMES})
     
       SET(CLONE_MODDIR "${CMAKE_BINARY_DIR}/${SD_LIBDIR}/drivers/${CLONENAME}")
-- 
2.54.0

